TestContract.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.4.16 <0.9.0;
+
+// Example contract of a TestContract.
+contract TestContract {
+
+    bool solved;    // Boolean to denote if contract is solved.
+    address payable owner;  // Owner of the contract, first this is the sponser.
+    uint expiry;        // Get the time when the contract expires.
+
+    constructor() payable {      // Constructor to initialise values.
+        solved = false;         //  Boolean value to indicate if contract is already solved.
+        owner = msg.sender;     //  Set the owner of the contract to the creator of the contract.
+        expiry = 1612569800;    //  Unix timestamp of the moment of expiry. 
+    }
+
+    function test(address payable hunter) public payable {
+        TemplateSolveContract solver = TemplateSolveContract(msg.sender); // The message sender is the contract activating the test function.
+        uint x = 100;                                   // Sample input.
+        uint16 y = 10;                                  // Sample expected output.
+        require(y == solver.main(x), "Wrong output");   // Require the output of the main function to be y.
+        solved = true;                                  // Set solved to true.
+        owner = hunter;                                 // Set the owner to the solver.
```

Comment:
for clarity, perhaps comment  'transfer ownership to the bounty hunter'. (instead of 'solver', which u also use as variable name above)

Reply:
+ in line 24 you use 'hunter' to refer to the person who solves, to avoid confusion it may be good to be consistent with how we refer to all parties involved :)